it/src/main/java/com/google/cloud/teleport/it/spanner/DefaultSpannerResourceManager.java

```diff
@@ -83,7 +85,7 @@ private DefaultSpannerResourceManager(Builder builder) {
       testId = generateNewId(testId, MAX_BASE_ID_LENGTH);
     }
     this.projectId = projectId;
-    this.instanceId = generateInstanceId(testId);
+    this.instanceId = generateInstanceId(testId, ILLEGAL_INSTANCE_CHARS, MAX_BASE_ID_LENGTH);
```

Comment:
I don't like the fact that the ID generation logic is now spread all over the place: `generateInstanceId` is in `ResourceManagerUtils`, `generateDatabaseId` and `ILLEGAL_DATABASE_CHARS` are in `SpannerResourceManagerUtils`, `ILLEGAL_INSTANCE_CHARS` is here and in `DefaultBigQueryResourceManager`. It makes it hard to answer the question "how do we generate Spanner instance and database ID" for example.

I suggest re-working this and extracting as many generic method is possible.
Maybe something like this?

1) Keep common methods that just generate arbitrary IDs based on the source testId  in `ResourceManagerUtils` (with some options like limit the length to N).
2) All Spanner ID generation logic should be in `SpannerResourceManagerUtils` (move `generateInstanceId` back there, even if it's going to be just a 1-liner that calls `ResourceManagerUtils`). So both `generateInstanceId` and `generateDatabaseId` should be there.
3) All Bigtable ID generation logic should be in `BigtableResourceManagerUtils` (create new class).
4) All constants like  `ILLEGAL_DATABASE/INSTANCE_CHARS` and "rules" (max length, allowed chars, whether to auto-replace some chars, etc.) of how the IDs have to be generated for a specific resource should be only in the `XResourceManagerUtils` class.
5) Consider renaming `ResourceManagerUtils` to `ResourceIdUtils` as I believe this ID generation logic may be used outside of Resource Managers in the future.

Reply:
I agree with points 1-4, however, I'm not sure about point 5. This class is meant to contain common functions used between resource managers, which just so happens to be generating ID's at this point. But I figured there will be other types of common functions contained in this class. Do you think there should be a separate class just for ID generation aside from other common functions?